src/components/tile.tsx

```diff
@@ -1,45 +1,98 @@
-import React, { FC } from 'react';
+import React, { CSSProperties, FC, useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
 
 import '../styles/components/tile.scss';
+import useSize from '../hooks/useSize';
 
 type Props = {
-  /**
-   * The namespace, which decides the colour
-   */
-  namespace?:
-    | 'uniref'
-    | 'uniparc'
-    | 'proteomes'
-    | 'uniprotkb'
-    | 'keywords'
-    | 'publications';
   /**
    * The tile title
    */
   title: string;
   /**
-   * The tile description. Short and snappy.
+   * The tile subtitle
+   */
+  subtitle?: string;
+  /**
+   * The tile description.
+   */
+  description?: JSX.Element;
+  /**
+   * The background color
+   */
+  backgroundColor?: string;
+  /**
+   * The background image
    */
-  description?: string;
+  backgroundImage?: JSX.Element;
   /**
-   * Small or normal version
+   * Whether to create a gradient based on the backgroung color or not
    */
-  small?: boolean;
+  gradient?: boolean;
+  /**
+   * The length of a side of the Tile square (css value). By default it will use the
+   * width of the provided container.
+   */
+  sideLength?: string;
+  /**
+   * Action triggered on click
+   */
+  onClick: () => void;
 };
 
 export const Tile: FC<Props> = ({
-  namespace,
   title,
-  description = 'This is a short description of what the resource is/provides.',
-  small,
+  subtitle,
+  description,
+  backgroundColor,
+  backgroundImage,
+  gradient = false,
+  sideLength,
+  onClick,
 }) => {
-  let classNames = 'tile';
-  classNames += namespace ? ` tile--${namespace}` : '';
-  classNames += small ? ' tile--small' : '';
+  const classes = classNames({ tile: 'tile', 'tile-gradient': gradient });
+  const ref = useRef<HTMLElement>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [size, setSize] = useState(sideLength);
+
+  const [sizes] = useSize(containerRef);
+
+  useEffect(() => {
+    if (sideLength) {
+      return;
+    }
+    if (sizes) {
+      setSize(sizes.width);
+    }
+  }, [sizes, sideLength]);
+
   return (
-    <div className={classNames}>
-      <h3 className="tile__header">{title}</h3>
-      {!small && <p className="tile__content">{description}</p>}
+    <div ref={containerRef}>
+      <div
+        className={classes}
+        style={
+          {
+            '--tile-background': backgroundColor,
+            width: size,
+            height: size,
+          } as CSSProperties
+        }
+        onClick={onClick}
+        role="button"
```

Comment:
I'd rather use a proper button element here instead of having to add role, tabIndex, and having to handle onKeyPress (if it's a button, onClick will handle everything)